src/WatchDogs.FakeSource/DataGenerator.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Bogus;
+
+namespace WatchDogs.FakeSource;
+
+//Deal #1, Balance 10 000, Buy EURUSD 1 lot at 2019-05-12 14:43:12
+//Deal #2, Balance 10 000, Sell GBPUSD 0.2 lots at 2019-05-12 14:43:23
+//Deal #3, Balance 1 000, Sell GBPUSD 1.2 lots at 2019-05-12 14:43:23
+//Deal #4, Balance 10 000, Sell GBPUSD 0.21 lot at 2019-05-12 14:43:24 <- triggered match with deal #2
+//Deal #5, Balance 20 000, Sell GBPUSD 0.4 lot at 2019-05-12 14:43:24 <- triggered match with deal #2 and deal #4
+
+public class DataGenerator : IDataGenerator
+{
+    // This holds an information or set of rules for generating fake data for TradeModel
+
+    Faker<TradeModel> tradeModelFake;
+
+    public DataGenerator()
+    {
+        // This field is optional and it allows us to replicate the results (everytime we runs an application, the app will generate same results)
+
+        //Randomizer.Seed = new Random(123);
+
+        DateTimeOffset initialTimestamp = DateTimeOffset.Now;
+
+        tradeModelFake = new Faker<TradeModel>()
+            .RuleFor(u => u.DealsGuid, f => f.Finance.Random.Guid())
+            .RuleFor(u => u.Currency, GetRandomCurrencyPair)
+            .RuleFor(u => u.TimeStamp, (f, u) =>
+            {
+                initialTimestamp = initialTimestamp.AddMilliseconds(f.Random.Int(0, 5000)); // Due to assignment, Timestamps should be closer to each other and in chronological order
+                return initialTimestamp;
+            })
+            .RuleFor(u => u.Action, f => f.PickRandom<TradeAction>())
+            .RuleFor(u => u.Lot, f => Math.Round(f.Random.Decimal(), 2))
+            .RuleFor(u => u.AccountBalance, f => Math.Round(f.Finance.Random.Decimal(1, 10000), 2)
+            );
+    }
+
+    public TradeModel GenerateFakeTrade()
+    {
+        return tradeModelFake.Generate();
+    }
+
+    public IEnumerable<TradeModel> GenerateFakeTrades()
+    {
+        return tradeModelFake.GenerateForever();
+    }
+
+    public List<TradeModel> LoadFakeData()
```

Comment:
Now that is the issue. You are basically returning a batch of fake deals, which is of course totally fine, but I would consider changing the type to `IEnumerable<Trade>` since it is a public method.

Reply:
Done although I do not understand the connection.